src/sagemaker_huggingface_inference_toolkit/optimum_utils.py

```diff
@@ -0,0 +1,91 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import importlib.util
+import logging
+import os
+
+
+_optimum_neuron = False
+if importlib.util.find_spec("optimum") is not None:
+    if importlib.util.find_spec("optimum.neuron") is not None:
+        _optimum_neuron = True
+
+logger = logging.getLogger(__name__)
+
+
+def is_optimum_neuron_available():
+    return _optimum_neuron
+
+
+def get_input_shapes(model_dir):
+    """Method to get input shapes from model config file. If config file is not present, default values are returned."""
+    from transformers import AutoConfig
+
+    input_shapes = {}
+    input_shapes_available = False
+    # try to get input shapes from config file
+    try:
+        config = AutoConfig.from_pretrained(model_dir)
+        if hasattr(config, "neuron_batch_size") and hasattr(config, "neuron_sequence_length"):
+            input_shapes["batch_size"] = config.neuron_batch_size
+            input_shapes["sequence_length"] = config.neuron_sequence_length
+            input_shapes_available = True
+            logger.info(
+                f"Input shapes found in config file. Using input shapes from config with batch size {input_shapes['batch_size']} and sequence length {input_shapes['sequence_length']}"
+            )
+    except Exception:
+        input_shapes_available = False
+
+    # return input shapes if available
+    if input_shapes_available:
+        return input_shapes
+
+    # extract input shapes from environment variables
+    sequence_length = os.environ.get("HF_OPTIMUM_SEQUENCE_LENGTH", None)
+    if not int(sequence_length) > 0:
+        raise ValueError(
+            f"HF_OPTIMUM_SEQUENCE_LENGTH must be set to a positive integer. Current value is {sequence_length}"
+        )
+    batch_size = os.environ.get("HF_OPTIMUM_BATCH_SIZE", 1)
+    logger.info(
+        f"Using input shapes from environment variables with batch size {batch_size} and sequence length {sequence_length}"
+    )
+    return {"batch_size": int(batch_size), "sequence_length": int(sequence_length)}
+
+
+def get_optimum_neuron_pipeline(task, model_dir):
+    """Method to get optimum neuron pipeline for a given task. Method checks if task is supported by optimum neuron and if required environment variables are set, in case model is not converted. If all checks pass, optimum neuron pipeline is returned. If checks fail, an error is raised."""
+    from optimum.neuron.pipelines import NEURONX_SUPPORTED_TASKS, pipeline
+    from optimum.neuron.utils import NEURON_FILE_NAME
+
+    # check task support
+    if task not in NEURONX_SUPPORTED_TASKS:
+        raise ValueError(
+            f"Task {task} is not supported by optimum neuron and inf2. Supported tasks are: {NEURONX_SUPPORTED_TASKS.keys()}"
```

Comment:
Logging dict keys in this way results in output that's a little weird:
``` python
>>> x = {'a': 1, 'b': 2}
>>> x.keys()
dict_keys(['a', 'b'])
>>> print(f"x keys are: {x.keys()}")
x keys are: dict_keys(['a', 'b'])
```

May want to pass to `list(...)`.

Reply:
agreed